### PR TITLE
Feature/soft delete non existing jobs on startup

### PR DESCRIPTION
**Pull request by linkdotnet**
Implements the possibility to set the JobDefintions defined in the code as the single source of truth.
Every Job and trigger which could not be associated with a defined job will be soft deleted.

### CODE REVIEW

File: source/Jobbr.Server/JobRegistry/RegistryBuilder.cs [hunk 1]
```diff
@@ -148,17 +160,59 @@ private static void AddTrigger(IJobStorageProvider storage, JobTriggerBase trigg
...
+        }
 
-            var recurringTrigger = trigger as RecurringTrigger;
-            if (recurringTrigger != null)
+        private static IList<Job> GetUndefinedJobs(ICollection<string> allDefinedJobs, IEnumerable<Job> allJobsInStorage)
+        {
+            return allJobsInStorage.Where(job => !allDefinedJobs.Contains(job.UniqueName)).ToList();
+        }
+
+        private static IList<JobTriggerBase> GetTriggerOfJob(IEnumerable<long> jobIds, IJobStorageProvider storage)
```
Comment (olibanjoli) [inline source/Jobbr.Server/JobRegistry/RegistryBuilder.cs]: GetTriggersOfJobs

**Comment by olibanjoli** [comment]
looks good to me. but i am missing some tests 💃

**Comment by michaelschnyder** [comment]
Thank you very much for your valuable contribution @linkdotnet! This is definitively a feature that makes Jobbr better for additional scenarios.

Some thoughts that came to my mind while reading through the PR.
* The overall code looks good and has even improved older parts! +1
* Additional TestCases added, +1
* There are now two strategies how to handle differences between Definition and Database, but it's not so obvious in code. I would like to have these two strategies more visible, either by
** Explicitly create collections for Left, Right and Merged Jobs and iterate through them, applying different update/soft-delete mechanisms depending on the strategy - OR
** Implement both strategies completely independent from each other, only sharing the code that is used to compare Jobs or Triggers.
* I see the PR as an alternative to the current (incomplete strategy). Maybe we could formulate the strategy on a higher level, and create predefined settings for the ones we have now:
** Initial: AddOrUpdateJobs( AddJobs: Yes, UpdateJobs: Yes, RemoveJobs: No, AddTriggers: Yes, UpdateTrigger: Yes, DisableTruiggers: Yes, OmitJobRuns: No)
** New: DefinitiionAsMaster ( AddJobs: Yes, UpdateJobs: Yes, **DeactivateJobs: Yes**, AddTriggers: Yes, UpdateTrigger: Yes, DisableTruiggers: Yes, **OmitJobRuns: Yes**)
* We should also update the documentation regarding this two strategies. This will also help to identify additional test cases

Let's have a short call on Monday afternoon if possible?

**Comment by linkdotnet** [comment]
Because of Eastern I am currently not available and will be back on Tuesday in the office. Would be a call on Tuesday morning okay?

**Comment by michaelschnyder** [comment]
sure, will send you an invite!

**Pull request merged**